src/main/java/de/retest/recheck/cli/subcommands/Ignore.java

```diff
@@ -123,24 +120,43 @@ private boolean inputValidation( final boolean all, final File testReport, final
 		return true;
 	}
 
-	private boolean allDifferencesAreBlacklisted( final TestReport report ) {
-		boolean allDifferencesAlreadyListed = true;
-		for ( final SuiteReplayResult suiteReplayResult : report.getSuiteReplayResults() ) {
-			for ( final TestReplayResult testReplayResult : suiteReplayResult.getTestReplayResults() ) {
-				for ( final ActionReplayResult actionReplayResult : testReplayResult.getActionReplayResults() ) {
-					for ( final ElementDifference elementDifference : actionReplayResult.getAllElementDifferences() ) {
-						for ( final AttributeDifference attributeDifference : elementDifference
-								.getAttributeDifferences( ignoreApplier ) ) {
-							final Element element = elementDifference.getElement();
-							if ( !ignoreApplier.matches( element, attributeDifference ) ) {
-								allDifferencesAlreadyListed = false;
-								ignoreApplier.ignoreAttribute( element, attributeDifference );
-							}
-						}
-					}
-				}
+	/**
+	 * Ignores all differences from the given test report via {@link Ignore#ignoreApplier}.
+	 *
+	 * @param report
+	 *            The test report to look for differences.
+	 * @return <code>true</code> if new differences have been ignored, otherwise <code>false</code>.
+	 */
+	private boolean ignoreDifferences( final TestReport report ) {
+		return report.getSuiteReplayResults().stream() //
+				.flatMap( suiteReplayResult -> suiteReplayResult.getTestReplayResults().stream() ) //
+				.flatMap( testReplayResult -> testReplayResult.getActionReplayResults().stream() ) //
+				.flatMap( actionReplayResult -> actionReplayResult.getAllElementDifferences().stream() ) //
+				.map( this::ignoreElementDifference ) //
+				.reduce( false, Boolean::logicalOr );
+	}
+
+	private boolean ignoreElementDifference( final ElementDifference diff ) {
+		return diff.isInsertionOrDeletion() ? ignoreElement( diff ) : ignoreAttributeDifferences( diff );
+	}
+
+	private boolean ignoreElement( final ElementDifference diff ) {
+		if ( !ignoreApplier.matches( diff.getElement() ) ) {
+			ignoreApplier.ignoreElement( diff.getElement() );
+			return true;
+		}
+		return false;
+	}
+
+	private boolean ignoreAttributeDifferences( final ElementDifference diff ) {
```

Comment:
We could, if we want, follow the new style and further decompose this:

```java
private boolean ignoreAttributeDifferences( final ElementDifference diff ) {
	final Element element = diff.getElement();
	return diff.getAttributeDifferences( ignoreApplier ).stream() //
			.map( attributeDiff -> ignoreAttributeDifference( element, attributeDiff ) ) //
			.reduce( false, Boolean::logicalOr );
}

private boolean ignoreAttributeDifference( final Element element, final AttributeDifference attributeDiff ) {
	if ( !ignoreApplier.matches( element, attributeDiff ) ) {
		ignoreApplier.ignoreAttribute( element, attributeDiff );
		return true;
	}
	return false;
}
```

What do you think?

Reply:
I think we sacrafice a little bit of readability to gain a little bit of consistency. I would be fine with either but since you request it i guess i'm going with it.